state/spaces.go

```diff
@@ -256,6 +258,45 @@ func (st *State) SpaceByName(name string) (*Space, error) {
 	return &Space{st, doc}, nil
 }
 
+// RenameSpace renames the input space `fromName` to input `toName`.
+// settingsDelta are the delta for the controllerSettings collection.
+// newConstraints are the new constraints to be input into the database.
+// An error is returned if the space does not exist or if there was a problem
+// accessing its information.
+func (st *State) RenameSpace(fromName, toName string, settingsDelta settings.ItemChanges, newConstraints map[string]constraints.Value) error {
```

Comment:
We want to build a `ModelOperation` that is passed to state for execution rather than adding a method to state.
See https://github.com/juju/juju/pull/10630/files#diff-4124533149cc21076d3c7d640d9463acR199